fossas/spectrometer
# Adds mix.exs parsing tactic for elixir

# Overview

Adds additional tactic for elixir/mix.

## Acceptance criteria

- When mix command tool is not accessible, it should fallback to reporting dependencies from mix.exs file.
- When mix command tool is not accessible, direct dependencies can be reported from mix.exs file
- From mix.exs file, dependencies environment are correctly inferred

## Testing plan

- I tested across many example open source projects. 

Example Elixir Project:
- https://github.com/absinthe-graphql/absinthe

1. Clone a elixir project
2. Ensure `mix` command tool is unreachable or does not exist
3. Run `fossa analyze -o` against example project

## Risks

- N/A

## References

Works on https://github.com/fossas/team-analysis/issues/398

## Checklist
